Zemismart/__init__.py

```diff
@@ -98,36 +100,47 @@ def connect(self):
             self.disconnect()
             raise
 
-
     def handleNotification(self, handle, data):
         if handle == self.datahandle.getHandle() and data[0] == self.response_start_byte:
             if data[1] == self.get_battery_cmd[0]:
                 battery = data[7]
-                print("Got battery: " + str(battery))
                 self.save_battery(battery)
+                self.last_command_status = True
             elif data[1] == self.get_position_cmd[0]:
                 pos = data[5]
-                print("Got position: " + str(pos))
                 self.save_position(pos)
+                self.last_command_status = True
             elif data[1] == self.finished_moving_cmd[0]:
                 pos = data[4]
-                print("Finished moving to position: " + str(pos))
                 self.save_position(pos)
-    
+                self.last_command_status = True
+            elif data[1] == 0x41:  # notify position
+                pos = data[4]
+                self.save_position(pos)
+                self.last_command_status = True
+            elif data[1] in (self.set_position_cmd[0], self.pin_cmd[0], self.move_cmd[0]):
+                if data[3] == 0x5A:
+                    self.last_command_status = True
+                elif data[3] == 0xA5:
+                    self.last_command_status = False
 
     def login(self):
-       pin_data = bytearray(struct.pack(">H", self.pin))
-       self.send_Zemismart_packet(self.pin_cmd, pin_data)
+        pin_data = bytearray(struct.pack(">H", self.pin))
+        self.send_Zemismart_packet(self.pin_cmd, pin_data)
 
     def send_BLE_packet(self, handle, data, wait_for_notification_time=0):
         write_response = handle.write(bytes(data), withResponse=False)
         if wait_for_notification_time > 0:
-            self.device.waitForNotifications(wait_for_notification_time)
+            start_time = time.time()

```

Comment:
Since this waiting is now specific for Zemismart (because it is waiting for a specific notification and not just any), I think it would be better to move this waiting to `send_Zemismart_packet` and remove the `wait_for_notification_time` in this method.